Sources/Datadog/Core/Upload/RequestBuilder.swift

```diff
@@ -126,9 +127,25 @@ internal struct RequestBuilder {
         computedHeaders.forEach { field, value in headers[field] = value() }
 
         request.httpMethod = "POST"
-        request.allHTTPHeaderFields = headers
-        request.httpBody = data
 
+        if let body = Deflate.encode(data) {
+            headers[HTTPHeader.contentEncodingHeaderField] = "deflate"
+            request.httpBody = body
+        } else {
+            request.httpBody = data
+#if DD_SDK_ENABLE_INTERNAL_MONITORING
+            InternalMonitoringFeature.instance?.monitor.sdkLogger.warn(
+                "Failed to compress request payload",
+                attributes: [
+                    "url": url,
+                    "headers": headers,
+                    "base64-data": data.base64EncodedString()
```

Comment:
I'm not sure if we want to send `headers` and `base64-data` fields:
* `headers` - Since V2 (https://github.com/DataDog/dd-sdk-ios/pull/562) headers include client-token, so we will be sending foreign tokens to our org.
* `base64-data` - This means sending serialized event, which may include sensitive information, e.g. 1/ we can't send user name & email without requiring extra GDPR clause; e.g. 2/ we don't want to collect secrets from unredacted RUM Resource urls occurring in the payload. 

Unless I'm overlooking some troubleshooting scenario, IMO this could be enough:
* `url` - to know which feature has sent the request (log / trace / rum),
* `uncompressed-size` - to know the uncompressed size (`data.count`) of the request.

This way, if we observe compression warnings for `~100KB` sent from logging feature, that would mean sth actionable. IMO we don't need to know the exact data as we know the schema and size. How does it sound?